openapi-maven-plugin/src/main/java/io/github/kbuntrock/configuration/library/reader/JakartaRsReader.java

```diff
@@ -157,7 +153,7 @@ protected List<ParameterObject> readParameters(final Method originalMethod, fina
 				logger.debug("Parameter : " + parameter.getName());
 
 				ParameterObject paramObj = new ParameterObject(parameter.getName(),
-					genericityResolver.getContextualType(parameter.getParameterizedType(), method));
+					genericityResolver.resolve(clazz, parameter.getParameterizedType()));
```

Comment:
How about using `method.getDeclaringClass()`? Would that works?

Reply:
No, that was the previous way to resolve it and it was not working that well.

Previously, in presence of a inheritance chain, I firstly creating a reference "array" object. Starting from the child and going up each link ; saving for this link the parametrized name and the actual type corresponding to that name.

Then, when I was resolving a method parameter or response, I was reading this linked representation starting from the method declaring class, and going down till the actual type was resolved.

This was not exhaustive, I was putting aside a lot of cases in creating this reference object.

Turns out Guava is much better in this than me. I totally deleted my first horrible attempt to handle this manually, and I let the "TypeToken" resolver do its work.